mmengine/structures/instance_data.py

```diff
@@ -165,9 +172,8 @@ def __getitem__(self, item: IndexType) -> 'InstanceData':
             # More details in https://github.com/numpy/numpy/issues/9464
             item = item.astype(np.int64) if item.dtype == np.int32 else item
             item = torch.from_numpy(item)
-        assert isinstance(
-            item, (str, slice, int, torch.LongTensor, torch.cuda.LongTensor,
-                   torch.BoolTensor, torch.cuda.BoolTensor))
+        assert isinstance(item, (str, slice, int, LongTypeTensor.__args__,
+                                 BoolTypeTensor.__args__))
```

Comment:
```suggestion
        assert isinstance(item, (str, slice, int, typing.get_args(LongTypeTensor), typing.get_args(BoolTypeTensor))
```

Reply:
Thanks for your suggestion, I found out that "typing.get_args" method only enabled when python>= 3.8. Do we need compatibility with older versions ?

Reply:
We also need to support Python3.7.